### ISSUE TITLE
new file dialog - filename persists after selecting folder

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#1290]
click on sample1.html.  Now click on a folder you created (e.g. newfolder)  the folder location is selected, but it wants to create a file called sample1.html.  The filename probably shouldn't be carried over.  It probably doesn't make sense to populate the filename textbox with the filename of selected files either, since that's bound to result in an error condition.

**Comment by childsb** [comment]
This behavior is intentional / models windows dialogs behavior.

Typical windows resource dialog behaviors:
-Clicking a file sets the "file name" field to that clicked file
-Clicking a folder doesn't reset the file name field (maybe the user had typed in file1.html and wants to keep the name).

I've found these behaviors useful if i want to name something similar to another existing file (ie. click on Sample1.html, change the 1 to a 5 then save).

Maybe consult UI group since this is mater of opinion? Lowering priority and moving to m5.

**Comment by JonFerraiolo** [comment]
I'm with Brad on this one. Brad claims the Windows UI works in a particular way (and that matches my memory) and I just verified that the Mac also works that way. I suggest closing this bug. @peller is that OK?

**Comment by peller** [comment]
I don't think "because Windows works that way" is good justification :-P but I'll give in on the folder thing.

I think my real problem is that you shouldn't be able to select files in a new file dialog.  It's automatically an error if you hit ok.  I understand you can pattern after an existing name, but I find it awkward.  Is there a non-Windows example to follow?

**Comment by JonFerraiolo** [comment]
same behavior on mac

**Comment by peller** [comment]
please provide an example.  where's a new file dialog I can try which takes a filename?   I'm not talking about the folder issue anymore, just the fact that there's a filename there at all based on selection.

**Comment by JonFerraiolo** [comment]
I see this in Mac/Chrome, PowerPoint and Preview. Open an image in Preview then go to Save As dialog. With file system showing, click on various PNG, JPG or PDF files on your file system and you'll see that they become the current file name even though clicking Save would wipe out your old file. Notice that as you change folders, the filename field doesn't change. I'm 99% sure this is all standard Mac behavior.

**Comment by peller** [comment]
That's a Save As dialog... it is perfectly valid to select a file that exists for your save as filename.  It's **not** valid to do this for new.  New file dialogs on the Mac typically don't seem to request any filename.

Call me crazy, I believe that UI should guide you towards valid input, not error conditions.

**Comment by JonFerraiolo** [comment]
Ah, that's a more fundamental UI question about how our "New" commands should work. What @childsb has implemented matches SaveAs behavior found on both Windows and Mac, and we use the same logic across all of our file operations (New and Save mostly). 

Most desktop software does this for "New":
- They immediately create a new whatever (e.g., a document or a spreadsheet or a presentation) with a default file name. For Adobe, it's Untitled1 (.psd, .ai, .pdf, ...). For PowerPoint, its Presentation1 (.pptx).
- When you save one of these newly created files for the first time, they force you into the SaveAs dialog even if you issue a Save command (which normally doesn't show a dialog)

Maybe we can mimic this New behavior, too. But there are a lot of UI cases in Maqetta to take into account, so clearly such a UI change is too much to attempt for M4.

**Comment by peller** [comment]
Yes, that's the basis for my concern.  The most obvious fix to preserve the current UI would be to disable selection of files (or hide files entirely) and not provide a filename.  That's what Eclipse does.

Anything else would be more than a UI change.  I'm sure it would impact how we deal with the filesystem, temporary files, etc.

**Comment by JonFerraiolo** [comment]
The UI has been this way forever. We certainly don't need to rush in changes here for M4.

**Comment by jhpedemonte** [comment]
-> M6

**Comment by JonFerraiolo** [comment]
Too late and too risky to try to address this for M7. -> M8
